ga4gh/backend.py

```diff
@@ -782,21 +782,34 @@ def runSearchRequest(
         self.endProfile()
         return responseString
 
-    def runListReferenceBases(self, id_, requestArgs):
+    def runListReferenceBases(self, id_, request):
         """
         Runs a listReferenceBases request for the specified ID and
         request arguments.
         """
-        compoundId = datamodel.ReferenceCompoundId.parse(id_)
+        if not request:
+            request = protocol.ListReferenceBasesRequest()
+        else:
+            try:
+                request = protocol.fromJson(
+                    request,
+                    protocol.ListReferenceBasesRequest)
+            except protocol.json_format.ParseError:
+                raise exceptions.InvalidJsonException(request)
+        request.reference_id = id_
+        compoundId = datamodel.ReferenceCompoundId.parse(request.reference_id)
         referenceSet = self.getDataRepository().getReferenceSet(
             compoundId.reference_set_id)
-        reference = referenceSet.getReference(id_)
-        start = _parseIntegerArgument(requestArgs, 'start', 0)
-        end = _parseIntegerArgument(requestArgs, 'end', reference.getLength())
+        reference = referenceSet.getReference(request.reference_id)
+        start = _parseIntegerArgument(
+            protocol.toJsonDict(request), 'start', 0)
+        end = _parseIntegerArgument(
+            protocol.toJsonDict(request),
+            'end', reference.getLength())
         if end == 0:  # assume meant "get all"
             end = reference.getLength()
-        if 'pageToken' in requestArgs:
-            pageTokenStr = requestArgs['pageToken']
+        if request.page_token:
+            pageTokenStr = request.page_token
             if pageTokenStr != "":
```

Comment:
This has already been tested for being the empty string, you can just call ``start = _parsePageToken(pageTokenStr, 1)[0]``